lib/theme_check/language_server/document_link_engine.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module ThemeCheck
+  module LanguageServer
+    class DocumentLinkEngine
+      include PositionHelper
+      include RegexHelpers
+
+      def initialize(storage)
+        @storage = storage
+      end
+
+      def document_links(uri)
+        buffer = @storage.read(uri)
+        matches(buffer, PARTIAL_RENDER).map do |match|
```

Comment:
Is this file specific to `render` support?

Reply:
For now, yes. But we might want to add support for things that get piped to asset_url in the future.